pkg/testing/tools/artifacts_api.go

```diff
@@ -165,32 +167,55 @@ func (aac ArtifactAPIClient) GetVersions(ctx context.Context) (list *VersionList
 func (aac ArtifactAPIClient) GetBuildsForVersion(ctx context.Context, version string) (builds *VersionBuilds, err error) {
 	joinedURL, err := aac.composeURL(fmt.Sprintf(artifactsAPIV1VersionBuildsEndpoint, version))
 	if err != nil {
-		return
+		return nil, err
 	}
 
 	resp, err := aac.createAndPerformRequest(ctx, joinedURL)
 	if err != nil {
-		err = fmt.Errorf("getting builds for version %s: %w", version, err)
-		return
+		return nil, fmt.Errorf("getting builds for version %s: %w", version, err)
 	}
 
 	defer resp.Body.Close()
 	return checkResponseAndUnmarshal[VersionBuilds](resp)
 }
 
+// FindBuild returns a build of the given `version` that does not match the `excludeHash` commit hash
+// and starts with the `offset` index on the current list of builds.
+// If there are no builds matching these conditions, returns `ErrBuildNotFound`.
+// `offset` value is used when the consumer does not wish to request the latest build.
+func (aac ArtifactAPIClient) FindBuild(ctx context.Context, version, excludeHash string, offset int) (buildDetails *BuildDetails, err error) {
```

Comment:
[Blocker]
So if I understood this doc, and considering the artifactAPI return a list of builds where the first, index 0, is the most recent build, then I get how to use the offset. However I think either it could be a boolean `skipLatestBuild` if there isn't a reason to set offset to anything grater than 1. Or the doc should explain _the artifactAPI return a list of builds where the first, index 0, is the most recent build ..._. Without further explanation it's still cryptic why sometimes it's set to `1`, other times to `0` and never to another value.
Would you mind adding some more explanation?

Just as an example of what I mean:

  ```suggestion
  // FindBuild returns a build of the given `version` that does not match the
  // `excludeHash` commit hash. It searches for a matching build from the latest
  // to the oldest, starting the search at the `offset` index in the list of builds.
  // Setting `offset` to 0 includes all builds, 1 skips the latest, and so forth.
  // If there are no builds matching these conditions, returns `ErrBuildNotFound`.
  func (aac ArtifactAPIClient) FindBuild(ctx context.Context, version, excludeHash string, offset int) (buildDetails *BuildDetails, err error) {
  ```

Reply:
would naming it `FindLastBuildBut(offset, ...)` make it more readable for you @AndersonQ 
i think using offset gives us flexibility